src/reducers/versions.tsx

```diff
@@ -428,6 +431,64 @@ export const fetchVersionsList = ({
   };
 };
 
+type CreateInternalDiffParams = {
+  version: ExternalVersionWithDiff;
+  baseVersionId: number;
+  headVersionId: number;
+};
+
+// This function returns an array of objects compatible with `react-diff-view`.
+export const createInternalDiffs = ({
+  baseVersionId,
+  headVersionId,
+  version,
+}: CreateInternalDiffParams) => {
+  const GIT_STATUS_TO_TYPE: { [mode: string]: string } = {
```

Comment:
Should this be an `enum` somewhere to enforce the fact that `type` needs to be one of these strings? Currently `type` in `DiffInfo` is just defined as `string`.

Reply:
If we can do the exact same thing (i.e. not using a switch/case), then why not.

Reply:
Ah! There are string enums: https://www.typescriptlang.org/docs/handbook/enums.html 🏆

Reply:
So... enums don't work in declaration files apparently: https://lukasbehal.com/2017-05-22-enums-in-declaration-files/.

I tried and failed. `enum` is never exported. I introduced a union type instead and we get what we want in the end: type safety on `type`.

Reply:
> enums don't work in declaration files 

That's too bad. Oh well.